tmt/base.py

```diff
@@ -3649,10 +3658,15 @@ def from_spec(cls, spec: _RawLink) -> 'Link':
         # `spec` can be either a string, fmf id, or relation:target mapping with
         # a single key (modulo `note` key, of course).
 
-        # String is simple: if `spec` is a string, it represents a target,
-        # and we use the default relationship.
+        # String is simple: if `spec` is a string, it represents a [relation:]target,
+        # and we use the default relationship if relation is not specified. Note that
+        # target should not include ':', that is, the relation must be provided if
+        # target includes ':'.
         if isinstance(spec, str):
-            return Link(relation=Link.DEFAULT_RELATIONSHIP, target=spec)
+            parts = spec.split(':', maxsplit=1)
+            if len(parts) == 1:
+                return Link(relation=Link.DEFAULT_RELATIONSHIP, target=spec)
+            return Link(relation=cast(_RawLinkRelationName, parts[0]), target=parts[1])
```

Comment:
How do hou know this is safe? :) `foo:https://redhat.com` can easily slip through, `foo` is no relation we know and support. I’d recommend either a regex with alternatives, or making sure the first item is in the list of allowed relations, `Links._relations`.

Reply:
Hi @happz, currently I added  `Link._relation` to help to validate the link relation (i.e. to make sure the first item is in the list of allowed relations).  Since we have  `Links._relations` and `Link._relations`, should we define a global var to unify the `relations`?

Reply:
I'd keep just one of these, no global variable, `Links._relations` already exists, let's keep it around to make the patch smaller.

Reply:
Yes,  we can use `Links._relations` in `class Link`.